docs/source/reference-lowlevel.rst

```diff
@@ -1,44 +1,40 @@
-=======================================================
- Introspecting and extending Trio with ``trio.hazmat``
-=======================================================
+=========================================================
+ Introspecting and extending Trio with ``trio.lowlevel``
+=========================================================
 
-.. module:: trio.hazmat
+.. module:: trio.lowlevel
 
-.. warning::
-   You probably don't want to use this module.
-
-:mod:`trio.hazmat` is Trio's "hazardous materials" layer: it contains
-APIs useful for introspecting and extending Trio. If you're writing
-ordinary, everyday code, then you can ignore this module completely.
-But sometimes you need something a bit lower level. Here are some
-examples of situations where you should reach for :mod:`trio.hazmat`:
+:mod:`trio.lowlevel` contains low-level APIs for introspecting and
+extending Trio. If you're writing ordinary, everyday code, then you
+can ignore this module completely. But sometimes you need something a
+bit lower level. Here are some examples of situations where you should
+reach for :mod:`trio.lowlevel`:
 
 * You want to implement a new :ref:`synchronization primitive
   <synchronization>` that Trio doesn't (yet) provide, like a
   reader-writer lock.
 * You want to extract low-level metrics to monitor the health of your
   application.
-* You want to add support for a low-level operating system interface
-  that Trio doesn't (yet) expose, like watching a filesystem directory
-  for changes.
+* You want to use a low-level operating system interface that Trio
+  doesn't (yet) provide its own wrappers for, like watching a
+  filesystem directory for changes.
 * You want to implement an interface for calling between Trio and
   another event loop within the same process.
 * You're writing a debugger and want to visualize Trio's task tree.
 * You need to interoperate with a C library whose API exposes raw file
   descriptors.
 
-Using :mod:`trio.hazmat` isn't really *that* hazardous; in fact you're
-already using it – it's how most of the functionality described in
-previous chapters is implemented. The APIs described here have
-strictly defined and carefully documented semantics, and are perfectly
-safe – *if* you read carefully and take proper precautions. Some of
-those strict semantics have `nasty big pointy teeth
+You don't need to be scared of :mod:`trio.lowlevel`, as long as you
+take proper precautions. These are real public APIs, with strictly
+defined and carefully documented semantics. But, you should read
+carefully and take proper precautions. Some of those strict semantics
```

Comment:
Is the phrase repetition with line 28 intentional? Perhaps something like this could also work:

```
But, you should pay attention while using them and certainly read the documentation in detail.
```

Reply:
Good point, tweaked the wording a bit.